stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1493,48 +1530,86 @@ impl NakamotoBlock {
 
     /// Verify that if this block has a coinbase, that its VRF proof is consistent with the leader
     /// public key's VRF key. If there is no coinbase tx, then this is a no-op.
+    ///
+    /// If this is a shadow block, then the coinbase recipient must be the burn address.  The VRF
+    /// proof will only be checked to see that it is present, since this is required by epoch 3.x.
     pub(crate) fn check_coinbase_tx(
         &self,
-        leader_vrf_key: &VRFPublicKey,
+        mainnet: bool,
+        leader_vrf_key: Option<&VRFPublicKey>,
         sortition_hash: &SortitionHash,
     ) -> Result<(), ChainstateError> {
         // If this block has a coinbase, then verify that its VRF proof was generated by this
         // block's miner.  We'll verify that the seed of this block-commit was generated from the
         // parnet tenure's VRF proof via the `validate_vrf_seed()` method, which requires that we
         // already have the parent block.
         if let Some(coinbase_tx) = self.get_coinbase_tx() {
-            let (_, _, vrf_proof_opt) = coinbase_tx
+            let (_, recipient_opt, vrf_proof_opt) = coinbase_tx
                 .try_as_coinbase()
                 .expect("FATAL: `get_coinbase_tx()` did not return a coinbase");
             let vrf_proof = vrf_proof_opt.ok_or(ChainstateError::InvalidStacksBlock(
                 "Nakamoto coinbase must have a VRF proof".into(),
             ))?;
 
-            // this block's VRF proof must have ben generated from the last sortition's sortition
-            // hash (which includes the last commit's VRF seed)
-            let valid = match VRF::verify(leader_vrf_key, vrf_proof, sortition_hash.as_bytes()) {
-                Ok(v) => v,
-                Err(e) => {
+            if self.is_shadow_block() {
+                let Some(recipient) = recipient_opt else {
```

Comment:
You could simplify these two if statements into one I do believe by just doing

```rust
                // must be the standard burn address for this network
                let burn_addr = StacksAddress::burn_address(mainnet).to_account_principal();
                if Some(burn_addr) != recipient_opt {
                    warn!("Invalid shadow block: recipient does not burn");
                    return Err(ChainstateError::InvalidStacksBlock(
                        "Shadow block did not pay to burn address".into(),
                    ));
                }
```

Reply:
This has been simplified by refactoring the shadow block system in bdd5ea3